Move boot from asm to C code

pj1031999 opened the pull request:
Rewrite main part of early boot process from asm to C code (putted in kernel/rpi.c).

In next steps I want to move more rest of the code to C (mostly cpu_mailbox_wait).

In this PR mmu and translation table are fully initialized in C code.

pj1031999 committed:
Move clear_bss from ASM to C.

pj1031999 committed:
Move enable_cache to C code.

pj1031999 committed:
Function for invalidate TLB in C code.

pj1031999 committed:
Enable MMU in C code.

pj1031999 committed:
Setup tmp stack in C code.

pj1031999 committed:
Move el3_only to C code.

pj1031999 committed:
More inline assembly :)

pj1031999 committed:
Call function in function!

pj1031999 committed:
Setup TLB in C.

pj1031999 committed:
Move change_el to C.

pj1031999 committed:
el2_entry in C

pj1031999 committed:
Setup MMU in C code.

pj1031999 committed:
Use macros from aarch64reg.h.

cahirwpz commented:
Please provide a description for the PR. While writing it please elaborate what will be the next step.

pj1031999 committed:
Move everything what is possible from boot.S to rpi.c.

Split initialization into three phases:
-- arm64_init
-- platform_stack
-- platform_init
